src/VisualStudio/Core/Def/Implementation/PdbSourceDocument/PdbSourceDocumentOutputWindowLogger.cs

```diff
@@ -0,0 +1,126 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Composition;
+using Microsoft.CodeAnalysis.Editor.Shared.Extensions;
+using Microsoft.CodeAnalysis.Editor.Shared.Utilities;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.PdbSourceDocument;
+using Microsoft.VisualStudio.ComponentModelHost;
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.Shell.Interop;
+
+namespace Microsoft.VisualStudio.LanguageServices.Implementation.PdbSourceDocument
+{
+    [Export(typeof(IPdbSourceDocumentLogger)), Shared]
+    internal sealed class PdbSourceDocumentOutputWindowLogger : IPdbSourceDocumentLogger
+    {
+        [ImportingConstructor]
+        [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+        public PdbSourceDocumentOutputWindowLogger()
+        {
+        }
+
+        public void Clear()
+        {
+            OutputPane.Clear();
+        }
+
+        public void Log(string message)
+        {
+            OutputPane.WriteLine(message);
+        }
+
+        private class OutputPane
```

Comment:
Not sure I follow why this is an embedded class... it's only statically used. Could just remove that layer and move all members up

Reply:
Yeah, fair call. This is just a copy and paste, thank you for keeping me honest. Will clean it up :)